src/main/java/org/shanerx/tradeshop/commands/CommandRunner.java

```diff
@@ -104,7 +106,19 @@ public void help() {
                     .append("\n&6/tradeshop reload &c - Reloads plugin configuration files\n");
         }
 
-        sendMessage(sb.toString());
+	    CustomCommandHandler handler = CustomCommandHandler.getInstance();
+        while (handler.iter().hasNext()) {
```

Comment:
Is this a per command help message?

Reply:
Yes. The addon provides the help message (description + usage) and I am simply including it in the output for /ts help

Reply:
If we could do that for our own sub commands to that would be nice. I was already thinking about doing it so that people could use '/ts help addProduct' to get a cleaner description and usage and the reg help command could just list commands.

Reply:
I thought about that, but I did not see a reason to do it.
Firstly, our own, “official” commands are displayed exactly the same way; secondly, for users who are trying to figure out how to use a command, this seems to be the easiest way to relay the information.
Is having yet _another_ command with _another_ help message really worth the hassle?
I think there is a chance that it would complicate things for end users instead of making them simpler.

Reply:
Wdym? I am only extending the present help to include addon commands.
A per command help would most certainly be an elegant solution, but would it be necessary? What information apart from the description, the usage/args and the name of the potential addon that introduces it (all of which are already displayed) could we display in the per command help?

Reply:
Its not necessarily more information as much as a cleaner look, I really hate how our current help menu looks, it just feels too full. Also most of the time you use a help command its because you're looking for usage on a single command or you just cant remember what a command is called. so having a straight list for 'help' and usage on 'help <command>' would be better.

Reply:
Fair point.

So you are suggesting we only have the command and the description in help, and the usage explained in the specific help for the command? That could indeed be cleaner, hadn‘t thought about that.

Reply:
I was actually going more with just command, but command and description could be fine too

Reply:
I will take care of it tomorrow.